src/LiftType.hs

```diff
@@ -67,38 +68,42 @@ liftType =
         case tr of
             Con tyCon ->
                 mk tyCon
-            App trA trB ->
-                AppT (go trA) (go trB)
             Fun trA trB ->
                 ConT ''(->) `AppT` go trA `AppT` go trB
+            App trA trB ->
+                AppT (go trA) (go trB)
```

Comment:
Is there a difference between the `Fun` and `App` constructor cases here? If `Fun` were to translate to `InfixT (go trA) ''(->) (go trB)` then it might result in slightly different (maybe more readable?) TH. Not sure how useful that would be.

Reply:
I don't think there is a difference, except a tiny one in speed—the `Fun` pattern should avoid building an intermediate `App` `TypeRep`. I really don't know Template Haskell at all, so I can't answer your other question. `->` isn't all that special in this regard; you'd presumably want special treatment for type operators as well. Unfortunately, we can't get their fixity. A few years ago I suggested packaging fixity info into `TyCon`s so we could `show` `TypeRep`s better, but no one wanted to.

Reply:
I just checked... It doesn't seem to make a difference for `-ddump-splices`.